core/src/main/java/org/pentaho/reporting/platform/plugin/ParameterDependencyGraph.java

```diff
@@ -69,29 +70,25 @@ public ParameterDependencyGraph( final MasterReport report,
                                    final Map<String, Object> computedParameterValues ) {
     this.dependencyGraph = new LinkedHashMap<>();
     this.allParameterNames = new HashSet<>( reportParameter.keySet() );
-    this.noDependencyInformationAvailable =
-      !processDependentParameters( report, reportParameter, parameterContext,
-        new StaticDataRow( computedParameterValues ) );
+    processDependentParameters( report, reportParameter, parameterContext, new StaticDataRow( computedParameterValues ) );
   }
 
   /**
    * Test support ..
    *
-   * @param noDeps
    * @param allParameterNames
    */
-  ParameterDependencyGraph( boolean noDeps, String... allParameterNames ) {
+  ParameterDependencyGraph( String... allParameterNames ) {
     this.dependencyGraph = new LinkedHashMap<>();
     this.allParameterNames = new HashSet<>( Arrays.asList( allParameterNames ) );
-    this.noDependencyInformationAvailable = noDeps;
   }
 
-  public void setNoDependencyInformationAvailable( boolean noDependencyInformationAvailable ) {
-    this.noDependencyInformationAvailable = noDependencyInformationAvailable;
+  public void setDependencyInformationAvailable( boolean dependencyInformationAvailable ) {
```

Comment:
You might want to make this setter package local. Public stuff is callable from everywhere, and most importantly has strict rules on when we may change or remove it. So its better to not make things public unless you really really need it to be public. As this method is only used in the unit test, which sits in the same package, package local will do fine without carrying the burden of those wacky rules around deprecation.

Reply:
Agreed, will make this change.